src/runners/http_service_runner.ts

```diff
@@ -28,41 +32,62 @@ process.on('unhandledRejection', err => {
     }
 });
 
+if (require.main === module) {
+    (async () => {
+        const meshClient = new WSClient(config.MESH_WEBSOCKET_URI);
+        await runHttpServiceAsync({ meshClient }, config);
+    })().catch(error => logger.error(error));
+}
+
 /**
  * This service handles the HTTP requests. This involves fetching from the database
  * as well as adding orders to mesh.
+ * @param dependencies If no mesh client is supplied, the HTTP service will start without it.
+ *                     It will provide defaults for other params.
  */
-(async () => {
-    const connection = await getDBConnectionAsync();
-    const app = express();
+export async function runHttpServiceAsync(
+    dependencies: AppDependencies,
+    _config: { HTTP_PORT: string; ETHEREUM_RPC_URL: string },
+    _app?: core.Express,
+): Promise<Server> {
+    const app = _app || express();
     app.use(requestLogger());
-    const server = app.listen(config.HTTP_PORT, () => {
-        logger.info(`API (HTTP) listening on port ${config.HTTP_PORT}!\nConfig: ${JSON.stringify(config, null, 2)}`);
+    const server = app.listen(_config.HTTP_PORT, () => {
+        logger.info(`API (HTTP) listening on port ${_config.HTTP_PORT}!\nConfig: ${JSON.stringify(_config, null, 2)}`);
     });
-    const stakingDataService = new StakingDataService(connection);
-    // tslint:disable-next-line:no-unused-expression
-    new StakingHttpService(app, stakingDataService);
-    let meshClient;
+
+    // staking http service
+    const connection = dependencies.connection || (await getDBConnectionAsync());
```

Comment:
I agree, I'd rather it fail if not present than create if undefined.

That way there is consistently one place where all dependencies are created/initialized.

Reply:
Removed all of the fallback code. Now there is a `getDefaultAppDependencies` function which is used both for the runners and for index.ts. Defaults depend on the config so we can treat the config as source of truth.